<a href="https://colab.research.google.com/github/JoshuneArriaga/FCFM/blob/main/Ej_Limpieza_Equipo8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Práctica 1-Limpieza de datos**
### **Equipo 8**
### Grupo 001 Lunes 7:00pm-10:00pm
**Integrantes:**



> Brenda Sofía Hernández Hdz.          1801285


> Andrea Sarahi Gómez Ávila            2034174



> Joshune Juditht Arriaga Gómez        1853668


> Héctor Eleazar Escobedo              1745909


> Daniel Jair Martinez Ávila           1872137


















**Base de datos: Google Playstore**

**Url:**https://www.kaggle.com/lava18/google-play-store-apps

**a)Importar librerías y bases de datos**

Primero se importan las librerías necesarias para realizar el proceso de limpieza de datos

In [ ]:
import pandas as pd
import numpy as np
import json

Despues utilizamos la siguiente función para subir el archivo 

In [ ]:
from google.colab import files
files.upload()

Procedemos a llamar al archivo

In [ ]:

df = pd.read_csv("googleplaystore.csv")

**b)Visualización de los datos.**

Utilizamos la siguiente instrucción para saber el numero de filas y columnas

In [ ]:
df.shape

(10841, 13)

La base de datos contiene 10,841 filas y 13 columnas 

Utilizamos el comando head para visualizar los primeros 10 datos de cada columna.


In [ ]:
df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

Utilizamos el comando tail para visualizar los ultimos 10 datos de cada columna.

In [ ]:
df.tail(10)

App  ...         Android Ver
10831                       payermonstationnement.fr  ...          4.0 and up
10832                                       FR Tides  ...          2.1 and up
10833                                    Chemin (fr)  ...          2.2 and up
10834                                  FR Calculator  ...          4.1 and up
10835                                       FR Forms  ...          4.0 and up
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

c)Descripción básica de las columnas

Nombres de las columnas 

In [ ]:
df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

Tipos de datos 

In [ ]:
df.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

Datos nulos

In [ ]:
df.isnull().any()

App               False
Category          False
Rating             True
Reviews           False
Size              False
Installs          False
Type               True
Price             False
Content Rating     True
Genres            False
Last Updated      False
Current Ver        True
Android Ver        True
dtype: bool

Las columnas rating, type, content rating, current ver y android ver contienen valores nulos

d)Clasificación de datos

In [ ]:
df.Category.unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

Tenemos que en la columna category se encuentran las clasificaciones de las apps

Se agrupararan todas las apps de la categoría belleza

---



In [ ]:
Belleza = df[df['Category']=='BEAUTY']
Belleza.to_csv('AppsDeBelleza.csv', index=False)

Clasificación segun el tipo

Primero rellenaremos los datos nulos de la columna type

In [ ]:
Type = df.fillna(0)

In [ ]:
df.Type.unique()

array(['Free', 'Paid', 0, '0'], dtype=object)

Agruparemos las apps dependiendo de su tipo

In [ ]:
Free= df[df['Type']=='Free']
Free.to_csv('AppsGratis.csv', index=False)
Paid= df[df['Type']=='Paid']
Paid.to_csv('AppsDePaga.csv', index=False)

e)Cambiar nombres a las columnas

In [ ]:
df.rename(columns = {'Genres':'Generos'})

App  ...         Android Ver
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...        4.0.3 and up
1                                    Coloring book moana  ...        4.0.3 and up
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...        4.0.3 and up
3                                  Sketch - Draw & Paint  ...          4.2 and up
4                  Pixel Draw - Number Art Coloring Book  ...          4.4 and up
...                                                  ...  ...                 ...
10836                                   Sya9a Maroc - FR  ...          4.1 and up
10837                   Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                             Parkinson Exercices FR  ...          2.2 and up
10839                      The SCP Foundation DB fr nn5n  ...  Varies with device
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10841 rows x 13 columns]